cmake/Modules/ComputeCppCompilerChecks.cmake

```diff
@@ -38,10 +38,16 @@ if(MSVC)
       RESULT_VARIABLE ComputeCpp_STL_CHECK_RESULT
       ERROR_QUIET
       OUTPUT_QUIET)
-    if(NOT ${ComputeCpp_STL_CHECK_RESULT} EQUAL 0)
-      message(STATUS "Device compiler cannot consume hosted STL headers. Using any parts of the STL will likely result in device compiler errors.")
-    else()
-    message(STATUS "Device compiler does not meet certain STL version requirements. Disabling version checks and hoping for the best.")
+    if(NOT ${ComputeCpp_STL_CHECK_RESULT} EQUAL 0)      
+      message(STATUS "Device compiler cannot consume hosted STL headers. Using any parts of the STL will likely result in device compiler errors")
+      if (${MSVC_VERSION} GREATER_EQUAL 1927)
```

Comment:
I'm really not a fan of putting hardcoded error checks like this in CMake. It ties this to a specific (but potentially unknown) version of ComputeCpp, and this will hang around. People might copy it and therefore it won't get updated. Or, as I worry, we'll fix this and forget to update the check here.

The best way would be to force the failure in the testing code we're already running. Do you know which header causes the compilation failure? We'd then be able to add it at line 16, and actually not really need any other changes.

Maybe we should be adding all the headers that ComputeCpp includes in its own headers to the test file, since that is completely representative.

Reply:
> Do you know which header causes the compilation failure? We'd then be able to add it at line 16, and actually not really need any other changes.

It's not that simple. On Windows, the header changes depending on the selected toolset, that's why the failure is happening now. The xutility header is fine in version 14.26 but the device compiler can't parse xutility in 14.27.

I understand your concern and I don't like to have these hard-coded numbers either, but this was the only way I could think to warn users about this problem.

Reply:
What I mean is, if you add the headers that are transitively included by Codeplay's `CL/sycl.hpp` to the test file, it will in turn include `<xutility>`, and if that header is from the troublesome version then we'll get a failure as expected. We could maybe have a better message (perhaps pointing users to the support forum, or similar?) but the general idea in this file is "if this breaks your Visual Studio is too new for us".

Reply:
Ah right, in this case we don't need to do anything. The existing message will be showed, but just as a warning. I changed it to a fatal error because I almost didn't notice it. I will try to improve the message then

Reply:
So if we do add the std headers from `CL/sycl.hpp` and that compile fails, it's fine to mark it as a `FATAL_ERROR`, because literally just including the header will fail the build. If it requires the define we can downgrade it to a `WARNING`, with the caveat of it might still fail depending on what gets used.

Reply:
That's the thing, you don't need to include anything from sycl, just by using the device compiler to build this

```c++
#include <ios>
int main() {return 0;}
```

Will fail. I will simply update the message to a FATAL_ERROR then

Reply:
I see what you mean. Would you be able to update the list then fatal error if it can't compile? The reason I ask is I know we don't include `<ios>` in ComputeCpp user headers so it gives us a more accurate idea of whether it will work or not. If we're forcibly including a header which will always fail the compile in an otherwise empty file then I'm more happy with the FATAL_ERROR. Does that make sense?

Reply:
I've updated message to a fatal error and added some text explaining a possible solution for the problem